stock_prices/stock_prices.py

```diff
@@ -2,29 +2,23 @@
 
 import argparse
 
-
+#Iterative solution
+#Time complexity: O(n^2)
 def find_max_profit(prices):
-    lowestPrice = prices[0]
-    highestPrice = 0
-    profit = 0
-    maxProfit = 0
-    for price in prices:
-        if price < lowestPrice:
-            lowestPrice = price
-        elif price > highestPrice:
-            highestPrice = price
-        profit = highestPrice - lowestPrice
-        if profit > maxProfit:
-            maxProfit = profit
-
-    # print(f'highest price: {highestPrice}')
-    # print(f'lowest price:  {lowestPrice}')
-    # print(f'max profit: {maxProfit}')
-
-    return maxProfit
-
-
-find_max_profit([10, 7, 5, 8, 11, 9])
+  maxProfit = -50000000
+  for i in range(len(prices)): #O(n)
+    currentPrice = prices[i]
+    for j in range(i+1, len(prices)): #O(n)
+      nextPrice =  prices[j]
```

Comment:
This is great logic to your approach!, using nested for loops can receive max profits, but there is a certain way for it be optimized towards a better time complexity. 

A good foundation to begin with would having setting/declaring a minimum profit to compare towards your prices and also maximum profit.